math/overflow.go

```diff
@@ -29,6 +29,27 @@ func AddUint64WithErr(a, b uint64) (uint64, error) {
 	return s, ErrAdditionOverflow
 }
 
+// AddInt64 performs addition on uint64 and logs an error if the addition overflows
+func AddInt64(a, b int64) int64 {
```

Comment:
also, what is the use case here to have a function that "tries" to add 2 numbers but if it fails, it will trace log the error and return an int? This might give wrong results if overflowing occurs. Suggestion to remove all these wrapper functions

Reply:
The underlying reason is historical... Overflow protection has been added later, and in many places where it was added, the function could not return an error.

It's definitely possible to remove these functions and replace them with the error-returning equivalents, but that would require extensive changes.